pkg/commands/drift/statefiles/statefiles.go

```diff
@@ -353,6 +365,52 @@ func (c *DriftStatefilesCommand) actualStatefileUris(ctx context.Context, logger
 	return gotURIs, nil
 }
 
+func (c *DriftStatefilesCommand) emptyStateFiles(ctx context.Context, gcsURIs []string) ([]string, error) {
+	type Resource struct {
+		Empty bool
+		URI   string
+	}
+	w := workerpool.New[*Resource](&workerpool.Config{
+		Concurrency: maxConcurrentRequests,
+		StopOnError: true,
+	})
+	for _, u := range gcsURIs {
+		uri := u
+		if err := w.Do(ctx, func() (*Resource, error) {
+			empty, err := c.terraformParser.StateWithoutResources(ctx, uri)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get determine if state file URI has resources: %w", err)
+			}
+			return &Resource{*empty, uri}, nil
+		}); err != nil && !errors.Is(err, workerpool.ErrStopped) {
+			return nil, fmt.Errorf("failed to execute terraform resources task: %w", err)
+		}
+	}
+
+	results, err := w.Done(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute terraform resources tasks in parallel: %w", err)
+	}
+
+	var emptyStateFiles []string
+	errs := []error{}
```

Comment:
Any special reason to prefer a slice instead of just joining errors directly. Like

```go
var merr error
for _, r := range results {
  // ...
  merr = errors.Join(merr, err)
}
if merr != nil {
  return nil, fmt.Errorf("failed to execute terraform resource tasks in parallel: %w", merr)
}
```